### PR TITLE
Remove if branches in arange | fix(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Replace if with where for performance.

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -591,11 +591,8 @@ def _adjust_args_for_arange_int_dtype(
     end = op.Cast(end, to=FLOAT.dtype)
     step = op.Cast(step, to=FLOAT.dtype)
 
-    if start < zero:
-        start = op.Ceil(start)
-
-    if step < zero:
-        start = op.Floor(start)
+    start = op.Where(op.Less(start, zero), start, op.Ceil(start))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:594]: Reverse

Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:594]:
```suggestion
    start = op.Where(op.Less(start, zero), op.Ceil(start), start)
```

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -591,11 +591,8 @@ def _adjust_args_for_arange_int_dtype(
...
+    start = op.Where(op.Less(step, zero), start, op.Floor(start))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:595]:
```suggestion
    start = op.Where(op.Less(step, zero), op.Floor(start), start)
```

**Review by thiagocrepaldi — APPROVED** [review]

**Review by fatcat-z — APPROVED** [review]
LGTM!

**Pull request merged**